Sources/HummingbirdHTTP2/HTTP2Channel.swift

```diff
@@ -16,39 +16,56 @@ import HTTPTypes
 import HummingbirdCore
 import Logging
 import NIOCore
+import NIOHTTP1
 import NIOHTTP2
 import NIOHTTPTypes
 import NIOHTTPTypesHTTP1
 import NIOHTTPTypesHTTP2
 import NIOPosix
 import NIOSSL
+import NIOTLS
 
 /// Child channel for processing HTTP1 with the option of upgrading to HTTP2
-public struct HTTP2UpgradeChannel: HTTPChannelHandler {
+public struct HTTP2UpgradeChannel: ServerChildChannel {
+    typealias HTTP1ConnectionOutput = HTTP1Channel.Value
+    typealias HTTP2ConnectionOutput = NIOHTTP2Handler.AsyncStreamMultiplexer<HTTP2StreamChannel.Value>
     public struct Value: ServerChildChannelValue {
-        let negotiatedHTTPVersion: EventLoopFuture<NIONegotiatedHTTPVersion<HTTP1Channel.Value, (NIOAsyncChannel<HTTP2Frame, HTTP2Frame>, NIOHTTP2Handler.AsyncStreamMultiplexer<HTTP1Channel.Value>)>>
+        let negotiatedHTTPVersion: EventLoopFuture<NIONegotiatedHTTPVersion<HTTP1ConnectionOutput, HTTP2ConnectionOutput>>
         public let channel: Channel
     }
 
     /// HTTP2 Upgrade configuration
     public struct Configuration: Sendable {
+        /// Idle timeout, how long connection is kept idle before closing
+        public var idleTimeout: Duration?
+        /// Maximum amount of time to wait before all streams are closed after second GOAWAY
+        public var gracefulCloseTimeout: Duration?
+        /// Maximum amount of time a connection can be open
+        public var maxAgeTimeout: Duration?
         /// Configuration applieds to HTTP2 stream channels
         public var streamConfiguration: HTTP1Channel.Configuration
 
         ///  Initialize HTTP2UpgradeChannel.Configuration
         /// - Parameters:
-        ///   - additionalChannelHandlers: Additional channel handlers to add to HTTP2 connection channel
+        ///   - idleTimeout: How long connection is kept idle before closing
+        ///   - maxGraceCloseTimeout: Maximum amount of time to wait before all streams are closed after second GOAWAY
         ///   - streamConfiguration: Configuration applieds to HTTP2 stream channels
         public init(
+            idleTimeout: Duration? = nil,
+            gracefulCloseTimeout: Duration? = nil,
+            maxAgeTimeout: Duration? = nil,
             streamConfiguration: HTTP1Channel.Configuration = .init()
         ) {
+            self.idleTimeout = idleTimeout
+            self.gracefulCloseTimeout = gracefulCloseTimeout
             self.streamConfiguration = streamConfiguration
         }
     }
 
     private let sslContext: NIOSSLContext
     private let http1: HTTP1Channel
-    public var responder: HTTPChannelHandler.Responder { self.http1.responder }
+    private let http2Stream: HTTP2StreamChannel
+    let configuration: Configuration
```

Comment:
Does this configuration become `public`?

Reply:
For consistency with other channels it should be public